conformance/utils/config/timeout.go

```diff
@@ -0,0 +1,129 @@
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package config
+
+import "time"
+
+type TimeoutConfig struct {
+	// CreateTimeout represents the maximum time for a Kubernetes object to be created.
+	// Max value for conformant implementation: None
+	// Used by: MustApplyObjectsWithCleanup, MustApplyWithCleanup
+	CreateTimeout time.Duration
+
+	// DeleteTimeout represents the maximum time for a Kubernetes object to be deleted.
+	// Max value for conformant implementation: None
+	// Used by: MustApplyObjectsWithCleanup, MustApplyWithCleanup
+	DeleteTimeout time.Duration
+
+	// GatewayMustHaveAddress represents the maximum time for at least one IP Address has been set in the status of a Gateway.
+	// Max value for conformant implementation: None
+	// Used by: WaitForGatewayAddress
+	GatewayMustHaveAddress time.Duration
+
+	// GatewayStatusMustHaveListeners represents the maximum time for a Gateway to have listeners in status that match the expected listeners.
+	// Max value for conformant implementation: None
+	// Used by: GatewayStatusMustHaveListeners
+	GatewayStatusMustHaveListeners time.Duration
+
+	// GWCMustBeAccepted represents the maximum time for a GatewayClass to have an Accepted condition set to true.
+	// Max value for conformant implementation: None
+	// Used by: GWCMustBeAccepted
+	GWCMustBeAccepted time.Duration
+
+	// HTTPRouteMustNotHaveParents represents the maximum time for an HTTPRoute to have either no parents or a single parent that is not accepted.
+	// Max value for conformant implementation: None
+	// Used by: HTTPRouteMustHaveNoAcceptedParents
+	HTTPRouteMustNotHaveParents time.Duration
+
+	// HTTPRouteMustHaveCondition represents the maximum time for an HTTPRoute to have the supplied Condition.
+	// Max value for conformant implementation: None
+	// Used by: HTTPRouteMustHaveCondition
+	HTTPRouteMustHaveCondition time.Duration
+
+	// HTTPRouteMustHaveParents represents the maximum time for an HTTPRoute to have parents in status that match the expected parents.
+	// Max value for conformant implementation: None
+	// Used by: HTTPRouteMustHaveParents
+	HTTPRouteMustHaveParents time.Duration
+
+	// ManifestFetchTimeout represents the maximum time for getting content from a https:// URL.
+	// Max value for conformant implementation: None
+	// Used by: getContentsFromPathOrURL
+	ManifestFetchTimeout time.Duration
+
+	// NamespacesMustBeReady represents the maximum time for all Pods and Gateways in a namespaces to be marked as ready.
+	// Max value for conformant implementation: None
+	// Used by: NamespacesMustBeReady
+	NamespacesMustBeReady time.Duration
+
+	// RequestTimeout represents the maximum time for making an HTTP Request or fetching a Kubernetes object.
```

Comment:
I don't think we should include "fetching a Kubernetes object" here. I think "RequestTimeout" should be reserved for HTTP requests made with our roundtripper. The other timeouts could either have different names or just leave them unchanged/not configurable. The places this PR uses request timeout for fetching k8s objects are probably a bit broad, for example one of those timeouts actually covers the following flow:

- request a list of Gateways
- iterate through to confirm each Gateway in list has ready condition set to true
- request a list of Pod
- iterate through to confirm each Pod in list has ready condition set to true

There's a lot of variety in the other places where RequestTimeout is currently be used and I think it would be difficult to create a generic timeout that covered all of them reasonably well. I think I'd lean towards reverting these specific uses of RequestTimeout to not be configurable and leaving this config for HTTP request timeouts.

Reply:
Thanks. Done.